DNS/Client/RequestResolver/UdpRequestResolver.cs

```diff
@@ -14,27 +15,26 @@ public class UdpRequestResolver : IRequestResolver {
             this.fallback = new NullRequestResolver();
         }
 
-        public ClientResponse Request(ClientRequest request) {
+        public async Task<ClientResponse> Request(ClientRequest request) {
             UdpClient udp = new UdpClient();
             IPEndPoint dns = request.Dns;
 
             try {
                 udp.Client.SendTimeout = 5000;
                 udp.Client.ReceiveTimeout = 5000;
 
-                udp.Connect(dns);
-                udp.Send(request.ToArray(), request.Size);
+                await udp.SendAsync(request.ToArray(), request.Size, dns);                
 
-                byte[] buffer = udp.Receive(ref dns);
+                byte[] buffer = (await udp.ReceiveAsync()).Buffer;
```

Comment:
Yes, Connect() no longer exists. That was an odd method, since udp is by definition connectionless. From reading the documentation, as far as I could tell all it did was allow you to not specify the endpoint in subsequent requests.